modules/engine/src/testing/listeners.spec.ts

```diff
@@ -741,4 +745,127 @@ describe(testName, () => {
       });
     });
   });
+
+  describe("handleConditionalTransferCreation", () => {
+    beforeEach(() => {
+      stub(listeners, "isWithdrawTransfer").resolves(Result.ok(false));
+      chainService.getRegisteredTransferByDefinition.resolves(
+        Result.ok({
+          definition: AddressZero,
+          name: "foo",
+          encodedCancel: "0x",
+          resolverEncoding: "bar",
+          stateEncoding: "baz",
+        }),
+      );
+    });
+
+    it("should emit CONDITIONAL_TRANSFER_ROUTING_COMPLETE for the end recipient", async () => {
+      return new Promise((res, rej) => {
```

Comment:
Will returning a promise in a test actually properly work here? can you `await` it instead?

Reply:
This is the documented way to do it in Mocha. It's working also.

Reply:
Huh neat-o